### DISCUSSION TITLE
SFCC - Maintaining Order status

### DISCUSSION THREAD

**Discussion by amitkathpal123** [SalesforceCommerceCloud/sfcc-ci#439] [Team Posts]
Hi All,

We have situation where all the order statues will be saved in SFCC & will be directly coming from Warehouse management (no OMS).

Can you please point me to any documentation that explains standard order status change flow or recommendations are welcome.

For example, What will be Order, Shipping and Payment status when:
1. All items are shipped
2. Few are shipped, few cancelled
3. Few shipped, few returned
4. All returned
5. All canceled 
6. Few returned, few canceled
7. Few are shipped, few cancelled, few returned
 --- many more such scenarios can happen

Thanks,
Amit

**Comment by zlatinz** [comment]
What are your business requirements related to this?
After all the system should behave the way business expects it.
If the standard statuses do not cover the business use cases, you can extend the model using custom statuses.

Have you looked at the documentation of the statuses in https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/DWAPI/scriptapi/html/api/class_dw_order_Order.html ?

Reminder: SFCC B2C Platform (p.k.a. Demandware Digital) is not an OMS solution.

**Comment by sanjaygautampfsweb-zz** [comment]
Hi Guys,

Please refer below link for maintaining order status.
Its having complete detailed explanation.

https://xchange.demandware.com/message/83316#83316

Thanks
Sanjay

**Comment by amitkathpal123** [comment]
Thanks for responding.
Actually client din't provided detailed requirements and asking for recommendations.
I understand, SFCC is not OMS system but client does want to store all order statues in SFCC.

And I have looked at the documentations links but questions still remain. For example: What should be the Order, Shipping & Payment Status in case of - one item shipped, one cancelled and one one returned. 

Is it something specific to each client & let let to decide or there are standard recommendations?

**Comment by sanjaygautampfsweb-zz** [comment]
Hi Amit,

Please find below suggestion –


ð  For example: What should be the Order, Shipping & Payment Status in case of - one item shipped, one cancelled and one returned.

ð  Suggestion: This is something you need to customize (may vary based on client need) –

1)      Create one custom attribute at ProductLineItem, name something like item_status.

2)      Update Line Items ‘item_status’ attribute by value  either shipped, partial shipped, returned (via job)

3)      If all items are either in shipped, partial shipped, returned – you should update OrderStatus as “Completed”, Shipping status as “Shipped” and Payment Status as “Paid”.

4)      If Some items status are missing then, surely order status should be move to “in Progress”, Shipping status “Partial shipped”, Payment status as “Paid”.

Hope, it’s helpful.
Please notify, if you have any concern.

Thanks
Sanjay

**Comment by amitkathpal123** [comment]
Thanks Everyone. I am able to define order/shipment/payment statuses for all scenarios and client even have the option to choose a different status from allowed OOB status.
